Issues when building an image on Apple Si

**Is your feature request related to a problem? Please describe.**
```
500.1 gmake[2]: *** No rule to make target '/usr/lib/x86_64-linux-gnu/libblas.so.3', needed by 'SimCore/libSimCore_Generators.so'.  Stop.
500.1 gmake[1]: *** [CMakeFiles/Makefile2:1835: SimCore/CMakeFiles/SimCore_Generators.dir/all] Error 2
```

and I also see a warning with
```
11.84 [  7%] Building CXX object ReconEvent/CMakeFiles/Recon_Event.dir/src/Recon/Event/HgcrocPulseTruth.cxx.o
12.12 In file included from /usr/include/c++/13/bits/stl_algo.h:61,
12.12                  from /usr/include/c++/13/functional:67,
12.12                  from /usr/local/include/root/TF1.h:25,
12.12                  from /code/Recon/include/Recon/Event/CompositePulse.h:6,
12.12                  from /code/Recon/include/Recon/Event/HgcrocPulseTruth.h:8,
12.12                  from /code/Recon/src/Recon/Event/HgcrocPulseTruth.cxx:1:
12.12 /usr/include/c++/13/bits/stl_heap.h: In function 'constexpr void std::__adjust_heap(_RandomAccessIterator, _Distance, _Distance, _Tp, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<pair<double, double>*, vector<pair<double, double> > >; _Distance = long int; _Tp = pair<double, double>; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<ldmx::HgcrocPulseTruth::getMax() const::<lambda(const pair<double, double>&, const pair<double, double>&)> >]':
12.12 /usr/include/c++/13/bits/stl_heap.h:224:5: note: parameter passing for argument of type 'std::pair<double, double>' when C++17 is enabled changed to match C++14 in GCC 10.1
12.12   224 |     __adjust_heap(_RandomAccessIterator __first, _Distance __holeIndex,
12.12       |     ^~~~~~~~~~~~~
```

**Describe the solution you'd like**


The first one is the real issue, and it seems it's manually added in the GENIE here
https://github.com/LDMX-Software/ldmx-sw/blob/65c6b5c984707aee650121ca9316e7f21d6c0889/cmake/BuildMacros.cmake#L136-L138

**Describe alternatives you've considered**

We either add the BLAS to the Docker, or remove it's use.

The second one seems to be solved by moving to `ranges::sort`

OK you are right that it works with `just compile`

I do get the
```
In file included from /usr/include/c++/13/bits/stl_algo.h:61,
                 from /usr/include/c++/13/functional:67,
                 from /usr/local/include/root/TF1.h:25,
                 from /Users/blackmac/Docs/1Research/LDMX/LocalLDMXSWTry2/ldmx-sw/Recon/include/Recon/Event/CompositePulse.h:6,
                 from /Users/blackmac/Docs/1Research/LDMX/LocalLDMXSWTry2/ldmx-sw/Recon/include/Recon/Event/HgcrocPulseTruth.h:8,
                 from /Users/blackmac/Docs/1Research/LDMX/LocalLDMXSWTry2/ldmx-sw/Recon/src/Recon/Event/HgcrocPulseTruth.cxx:1:
/usr/include/c++/13/bits/stl_heap.h: In function 'constexpr void std::__adjust_heap(_RandomAccessIterator, _Distance, _Distance, _Tp, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<pair<double, double>*, vector<pair<double, double> > >; _Distance = long int; _Tp = pair<double, double>; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<ldmx::HgcrocPulseTruth::getMax() const::<lambda(const pair<double, double>&, const pair<double, double>&)> >]':
```
like that too.

And then later on I get the other one too:
```
gmake[2]: *** No rule to make target '/usr/lib/x86_64-linux-gnu/libblas.so.3', needed by 'SimCore/libSimCore_Generators.so'.  Stop.
gmake[2]: *** Waiting for unfinished jobs....
```

--- a/.gitignore
+++ b/.gitignore
@@ -31,3 +31,4 @@ EventDisplay/include/EventDisplay/EventDisplayLinkDef.h
 .bash*
 .profile
 .denv
+.apptainer

--- a/Recon/src/Recon/Event/HgcrocPulseTruth.cxx
+++ b/Recon/src/Recon/Event/HgcrocPulseTruth.cxx
@@ -9,7 +9,7 @@ void HgcrocPulseTruth::clear() {}
 double HgcrocPulseTruth::getMax() const {
   auto hits = compositePulse_.hits();
 
-  std::sort(
+  std::ranges::sort(
       hits.begin(), hits.end(),
       [](const std::pair<double, double> &a,
          const std::pair<double, double> &b) { return a.second < b.second; });

--- a/cmake/BuildMacros.cmake
+++ b/cmake/BuildMacros.cmake
@@ -133,10 +133,6 @@ macro(setup_genie_target)
     add_genie_target(GRwIO)
     add_genie_target(GRwClc)
     
-    add_library(blas SHARED IMPORTED GLOBAL) # or STATIC instead of SHARED
-    set_target_properties(blas PROPERTIES
-      IMPORTED_LOCATION "/usr/lib/x86_64-linux-gnu/libblas.so.3")
-    
     set(GENIE_LIBS
        log4cpp
        gsl

--- a/cmake/BuildMacros.cmake
+++ b/cmake/BuildMacros.cmake
@@ -193,8 +193,7 @@ macro(setup_genie_target)
        Genie::GTlGeo
        Genie::GTlFlx
        ROOT::MathCore
-       ROOT::MathMore
-       blas)
+       ROOT::MathMore)
      message(STATUS "Found GENIE")
   endif()
 endmacro()

